pkg/api/validation/validation.go

```diff
@@ -67,26 +68,6 @@ var PortNameErrorMsg string = fmt.Sprintf(`must be an IANA_SVC_NAME (at most 15
 
 const totalAnnotationSizeLimitB int = 256 * (1 << 10) // 256 kB
 
-func ValidateLabelName(labelName string, fldPath *field.Path) field.ErrorList {
```

Comment:
There was an import cycle in implementing validation for PodAffinityTerm:
1. `ValidateLabelSelector()` is in `api/unversioned/validation`, which imports `api/validation`, to access `ValidateLabels()`
2. `ValidatePodAffinityTerms()` is `api/validation`
3. To implement `ValidatePodAffinityTerms()`, we need to import `api/unversioned/validation` to access `ValidateLabelSelector()`, then the import cycle occurs -- `api/unversioned/validation` imports `api/validation` while `api/validation` itself imports `api/unversioned/validation`.

Suggestion from @davidopp and @lavalamp :  
> The general principle is that unversioned stuff should not depend on versioned stuff. Unversioned is like the root of a tree. For example, (unversioned) validation should not depend on apivalidation, and the types in "api" (and the code needed to implement it) should not depend on the types in "v1" (and the code needed to implement it). The reverse is OK (and expected) -- versioned stuff can depend on unversioned. There is a comment at the top of pkg/api/unversioned/types.go that says "The package contains two categories of types:
- external (serialized) types that lack their own version (e.g TypeMeta)
- internal (never-serialized) types that are needed by several different api groups, and so live here, to avoid duplication and/or import loops (e.g. LabelSelector)."

> So, it is wrong that api/unversioned/validation currently imports api/validation. We should fix this. Daniel's suggestion was to move apivalidation.ValidateLabels() and apivalidation.ValidateLabelName() into (unversioned) validation, so that api/unversioned/validation no longer needs to import api/validation.